DynamicIntArray.cpp

```diff
@@ -0,0 +1,90 @@
+#include "DynamicIntArray.h"
+#include <iostream>
+
+DynamicIntArray::DynamicIntArray(): m_size(0), m_array(new int[0] {}) {};
```

Comment:
new int[0] is invalid operation

Reply:
```
3.7.3.1/2
The effect of dereferencing a pointer returned as a request for zero size is undefined.
```
So what is the point to allocate memory to smth you can't properly use? In your case there's no real benefit